zio-http/src/main/scala/zio/http/codec/RichTextCodec.scala

```diff
@@ -154,15 +154,17 @@ sealed trait RichTextCodec[A] { self =>
 }
 object RichTextCodec {
   private[codec] case object Empty                                        extends RichTextCodec[Unit]
-  private[codec] final case class CharIn(set: BitSet)                     extends RichTextCodec[Char]
+  private[codec] final case class CharIn(set: BitSet)                     extends RichTextCodec[Char] {
+    lazy val errorMessage = s"Not found: ${set.toArray.map(_.toChar).mkString}"
```

Comment:
You can do one better by removing `lazy`, and adding:

```scala
private[codec] val leftErrorMessage = Left(errorMessage)
```

Then using `leftErrorMessage` below.